### PR TITLE
subworkflow: fasta_align_index

### PR DESCRIPTION
**Pull request by matthdsm**
Uber aligner index generating workflow to go with `fastq_align_dna`

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by matthdsm** [comment]
This subworkflow expands a little on the common usage of the index for `bwa` and `bwamem2`. Most of the time the usage of the `altliftover` file is omitted, but this causes a LOT of false negatives when using hg38.
Indexes generated by this workflow (when adding the `altliftover` file) should fix that (especially in the case of `bwamem2`)
cfr: https://github.com/bwa-mem2/bwa-mem2/issues/208

**Comment by maxulysse** [comment]
Shouldn't it be fasta_index_dna ? To reflect more the relationship with fastq_align_dna ?

**Comment by matthdsm** [comment]
No opinion, I can change to whatever you like 😃 `fasta_index_dna` does seem better though

**Comment by matthdsm** [comment]
Fixed the name. Any more comments?

**Comment by SPPearce** [comment]
It is not obvious to me what `altliftover` option should be.
Can you provide some tests where it is not empty?

**Comment by matthdsm** [comment]
There's no test data available that would make sense in this context. I did include links and some explanation in `meta.yaml`

File: subworkflows/nf-core/fasta_index_dna/meta.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+      type: map
+      description: |
+        Groovy Map containing altliftover information
+        e.g. [ id:'test' ]
+  - altliftover:
+      type: file
+      description: |
+        Sam formatted liftover file for the reference genome alt contigs
+        See: https://github.com/lh3/bwa/blob/master/README-alt.md
+        Download from: https://sourceforge.net/projects/bio-bwa/files/bwakit
```
Comment (matthdsm) [inline subworkflows/nf-core/fasta_index_dna/meta.yml:37]: @SPPearce, see here

Comment (maxulysse) [reply subworkflows/nf-core/fasta_index_dna/meta.yml:37]: Only used for snap then?

Comment (matthdsm) [reply subworkflows/nf-core/fasta_index_dna/meta.yml:37]: Only used by `snap` during index generation. `bwamem2` reads the file as is and `bwa` uses it in the `bwakit` postalt script AFAIK

**Review by maxulysse — APPROVED** [review]
Why not using it for the fastq_align_dna tests?

**Comment by matthdsm** [comment]
> Why not using it for the fastq_align_dna tests?

There's no al file for sarsCov AFAIK (no alt chr) and the test data for human is only chr22, so not test data also. Only way to really test is full size data.

**Review by nvnieuwk — APPROVED** [review]
LGTM

**Pull request merged**